.github/workflows/test.yaml

```diff
@@ -0,0 +1,52 @@
+name: SPDK E2E TEST
+
+on:
+  workflow_dispatch:
+    inputs:
+      uuid:
+        description: ''
+        required: true
+        default: '79276661-5f8a-405d-ab6d-651b88326206'
+      ip:
+        description: ''
+        required: true
+        default: '18.218.243.112'
+      secret:
+        description: ''
+        required: true
+        default: '8K9BVlNfYo6aiLamu1c2'
+jobs:
+  build:
+    name: Build
+    runs-on:  ubuntu-latest
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v5
+      with:
+        go-version: ^1.21
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v4
+      
+    - name: Install kubectl
+      run: |
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+        sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+      
+    - name: Authenticate with Kubernetes cluster
+      run: |
+        mkdir -p "${HOME}/.kube"
+        echo ${{ secrets.KUBE_CONFIG_DATA }} | base64 --decode >  ${HOME}/.kube/config
```

Comment:
kubeconfig data also changes for a new cluster. You might also have to take kubeconfig from the workflow dispatch.

Reply:
I think leaving it as a secret is much better as it needs to be treated as one.
It is easily updated by changing the secret value with the new `kubeconfig`

Reply:
No we don't.
we only update it if the cluster we are testing on changes.